ui/app/src/views/ViewDashboardList.tsx

```diff
@@ -16,19 +16,57 @@ import {
   AccordionDetails,
   AccordionSummary,
   Box,
+  Button,
   CircularProgress,
   Container,
+  IconButton,
   Stack,
   Typography,
 } from '@mui/material';
+
+import { MouseEvent, useState } from 'react';
 import { DashboardResource } from '@perses-dev/core';
 import { ErrorAlert, ErrorBoundary } from '@perses-dev/components';
 import ChevronDown from 'mdi-material-ui/ChevronDown';
 import FolderPound from 'mdi-material-ui/FolderPound';
+import TrashCan from 'mdi-material-ui/TrashCan';
+import { useNavigate } from 'react-router-dom';
 import { useDashboardList } from '../model/dashboard-client';
+import { useCreateProjectMutation, useDeleteProjectMutation, ProjectModel } from '../model/project-client';
+import { useSnackbar } from '../context/SnackbarProvider';
 import DashboardList from '../components/DashboardList';
+import DeleteProjectDialog from '../components/DeleteProjectDialog/DeleteProjectDialog';
+import CreateProjectDialog from '../components/CreateProjectDialog/CreateProjectDialog';
 
 function RenderDashboardList() {
+  const [openDeleteProjectDialog, setOpenDeleteProjectDialog] = useState<boolean>(false);
+  const [projectToDelete, setProjectToDelete] = useState<string | undefined>(undefined);
+
+  const openDeleteProjectConfirmDialog = ($event: MouseEvent, name: string) => {
+    $event.stopPropagation(); // Preventing the accordion to toggle when we click on the button
+    setProjectToDelete(name);
+    setOpenDeleteProjectDialog(true);
+  };
+
+  const closeDeleteProjectConfirmDialog = () => {
+    setOpenDeleteProjectDialog(false);
+  };
+
+  const { successSnackbar, exceptionSnackbar } = useSnackbar();
+  const deleteProjectMutation = useDeleteProjectMutation();
+  const triggerProjectDeletion = function () {
+    if (projectToDelete) {
+      deleteProjectMutation.mutate(projectToDelete, {
+        onSuccess: (name: string) => {
+          successSnackbar(`project ${name} was successfully deleted`);
+        },
+        onError: (err) => {
+          exceptionSnackbar(err);
+        },
+      });
+    }
+  };
```

Comment:
I wonder if we want to move some of this logic for calling the API to delete the project into the DeleteProjectDialog? 

My thinking is that it would make this file smaller, and it would keep more of the code related to deleting a project in one place.

Same for the code related to calling the API to create a project, if this idea sounds nice (we could put that logic in the CreateProjectDialog).

Reply:
I wondered the same thing! I created first a TODO to think about it and then as everything was working well I just passed. 
There is also this wishful thinking of dialogs components for consistency https://github.com/perses/perses/discussions/935 that would be a bit more complex if the business logic was inside the dialog.

Still having it inside the dialog will bring clarity on this code and you convinced me so I'm gonna do it :)

Reply:
@cndonovan updated! I added the logic into ``DeleteProjectDialog``/``AddProjectDialog`` and it's faar more elegant and clear.

I also discovered the ``useCallback`` which was advised by my IDE's Sonar locally. So I know a bit more about rendering and that should optimize slightly the computation. 

I also changed the delete icon to the outlined Delete, in order to use the same as in dashboards.